Dockerfile.aaq-controller

```diff
@@ -1,33 +1,37 @@
 # Builder stage
-FROM docker.io/library/golang:1.22-alpine as builder
+FROM  --platform=${BUILDPLATFORM} docker.io/library/golang:1.23-alpine as builder
+ARG BUILDPLATFORM
 
 # Install make
 RUN apk update && apk add make
 
-# Copy the source code from the host to the container
-COPY pkg /workdir/app/pkg
-COPY staging /workdir/app/staging
-COPY cmd /workdir/app/cmd
-COPY vendor /workdir/app/vendor
-COPY go.mod /workdir/app/go.mod
-COPY go.sum /workdir/app/go.sum
-COPY Makefile /workdir/app/Makefile
-
 WORKDIR /workdir/app
 
-RUN make aaq_controller
+# Copy the source code from the host to the container
+COPY pkg ./pkg
+COPY staging ./staging
+COPY cmd ./cmd
+COPY vendor ./vendor
+COPY go.mod ./go.mod
+COPY go.sum ./go.sum
+COPY Makefile ./Makefile
+
+ARG TARGETOS
+ARG TARGETARCH
+RUN GOOS=${TARGETOS} GOARCH=${TARGETARCH} make aaq_controller
 
 # Final stage
-FROM docker.io/library/golang:1.22-alpine
-
-# Copy the binary from the builder stage to the final image
-COPY --from=builder /workdir/app/aaq_controller /app/aaq_controller
+FROM --platform=${TARGETPLATFORM} docker.io/alpine
+ARG TARGETPLATFORM
 
 # Set the working directory to /app
-WORKDIR /app
+WORKDIR /app/aaq_controller
```

Comment:
What is the reasonbehind adding the binary name as a directory and then moving the binary inside it?